src/Component/AutoMapper/Generator/Generator.php

```diff
@@ -149,6 +149,15 @@ public function generate(MapperGeneratorMetadataInterface $mapperGeneratorMetada
             $sourcePropertyAccessor = $propertyMapping->getReadAccessor()->getExpression($sourceInput);
             [$output, $propStatements] = $transformer->transform($sourcePropertyAccessor, $result, $propertyMapping, $uniqueVariableScope);
 
+            $extractCallback = $propertyMapping->getReadAccessor()->getExtractCallback($mapperGeneratorMetadata->getSource());
+
+            if (null !== $extractCallback) {
+                $constructStatements[] = new Stmt\Expression(new Expr\Assign(
+                    new Expr\ArrayDimFetch(new Expr\PropertyFetch(new Expr\Variable('this'), 'extractCallbacks'), new Scalar\String_($propertyMapping->getProperty())),
+                    $extractCallback
+                ));
+            }
```

Comment:
the `extractCallbacks` could be used with constructors (whereas `hydrateCallbacks` could not), and they do not need a `writeMutator`. So, I reordered the statements to create the `extractCallbacks` even if we don't have a write murator.

However, I'm wondering if we should condition this only if we have a `writeMutatorConstructor`... WDYT?